kaos/api/errors.py

```diff
@@ -0,0 +1,61 @@
+"""Defines all errors used by the view functions to signal a response to the caller."""
+
+from flask import jsonify
+
+class APIError(Exception):
+    """Base class for all API related errors which are designed to trigger responses to the users.
+
+    These errors are handled by the application and printed as a standard JSON response to
+    the user.
+    """
+
+    def __init__(self, reasons, status_code=400, **kwargs):
+        """Args:
+            reasons (dict): A dictionary containing the reason(s) for a failure.
+            **kwargs: Additional information that will be attached to the generated response.
+
+        Kwargs:
+            status_code (int): status code of the generated response. Defaults to 400.
```

Comment:
Put this under Args.

ref: https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html

Reply:
Oh, I see. I could have sworn I saw something about ```KWargs:```. Will fix